fmriprep/interfaces/multiecho.py

```diff
@@ -8,44 +8,157 @@
 
 Using multi-echo EPI data, generates a T2*-map
 for use in T2*-driven EPI->T1 coregistration
+
+Change directory to provide relative paths for doctests
+>>> import os
+>>> filepath = os.path.dirname( os.path.realpath( __file__ ) )
+>>> datadir = os.path.realpath(os.path.join(filepath, '../data/'))
+>>> os.chdir(datadir)
+
 """
 import os
 import numpy as np
 import nibabel as nb
+from nilearn.masking import (apply_mask, unmask)
 
 from niworkflows.nipype import logging
-from niworkflows.nipype.utils.filemanip import split_filename
+from niworkflows.nipype.utils.filemanip import (split_filename, fname_presuffix)
 from niworkflows.nipype.interfaces.base import (
     traits, TraitedSpec, File, InputMultiPath, SimpleInterface,
     BaseInterfaceInputSpec)
 
 LOGGER = logging.getLogger('interface')
 
 
+class FirstEchoInputSpec(BaseInterfaceInputSpec):
+    in_files = InputMultiPath(File(exists=True), mandatory=True, minlen=2,
+                              desc='multi-echo BOLD EPIs')
+    ref_imgs = InputMultiPath(File(exists=True), mandatory=True, minlen=2,
+                              desc='generated reference image for each '
+                              'multi-echo BOLD EPI')
+
+
+class FirstEchoOutputSpec(TraitedSpec):
+    first_image = File(exists=True,
+                       desc='BOLD EPI series for the first echo')
+    first_ref_image = File(exists=True, desc='generated reference image for '
+                                             'the first echo')
+
+
+class FirstEcho(SimpleInterface):
+    """
+    Finds the first echo in a multi-echo series and its associated reference
+    image.
+
+    Example
+    =======
+    >>> from fmriprep.interfaces import multiecho
+    >>> first_echo = multiecho.FirstEcho()
+    >>> first_echo.inputs.in_files = ['sub-01_run-01_echo-1_bold.nii.gz', \
+                                      'sub-01_run-01_echo-2_bold.nii.gz', \
+                                      'sub-01_run-01_echo-3_bold.nii.gz']
+    >>> first_echo.inputs.ref_imgs = ['sub-01_run-01_echo-1_bold.nii.gz', \
+                                      'sub-01_run-01_echo-2_bold.nii.gz', \
+                                      'sub-01_run-01_echo-3_bold.nii.gz']
+    >>> res = first_echo.run() # doctest: +SKIP
```

Comment:
Since this is only checking names, you can do:

```Python
>>> res = first_echo.run()  # removing the +SKIP flag
>>> res.outputs.first_image
'sub-01_run-01_echo-1_bold.nii.gz'
>>> res.outputs.first_ref_image
'sub-01_run-01_echo-1_bold.nii.gz'
```